cmd/agent/install_mac_os.sh

```diff
@@ -423,7 +430,10 @@ else
     # the GUI was not running for the user (e.g. a run of this script via
     # ssh for user not logged in via GUI).
     if $cmd_launchctl print "gui/$user_uid/$service_name" 1>/dev/null 2>/dev/null; then
-        $cmd_real_user osascript -e 'tell application "System Events" to if login item "Datadog Agent" exists then delete login item "Datadog Agent"'
+        # retro-compatibility : remove dd-agent tray app login item only if it was started
+        if [ ! -f "/tmp/install-ddagent/tray-not-started" ]; then
```

Comment:
I don't understand why we need this tray-not-started file. If we did a systemwide install, then we will never enter this condition, will we?

Reply:
Since the names are a bit confusing the way I interpreted it is that line 425 checks if there is a running datadog-agent process running in user mode then we'll remove the tray app from the login and unload the agent service. Meaning that in system-wide the agent is still run a first time in user-mode before being unloaded so I need to remove tray only when necessary.

if line 425 actually checks for the process of the tray app then yes we don't need the `tray-not-started` file.
But now that you point it out i'm not exactly sure which of the two it is 🤔

Reply:
The way this condition is supposed to work is that if there is an existing single-user installation, we can tell by seeing if there is a user-specific instance of datadog-agent via launchctl. If there isn't one, that means that the tray app doesn't exist, right? If we're doing systemwide install, then the logic of this PR is that we never start the tray app in the first place, so this file shouldn't be needed. Am I wrong?

Reply:
🤔 My understanding is that the change I applied on the `postinst` script only prevent the tray app to be started in systemwide installation. This condition is checking if there is a user-specific instance of datadog-agent, meaning that even in systemwide installation this condition was true since in both way (systemwide & single user) we had the 2 prompts. 

So I think that there is always an user specific instance of datadog-agent started on install but its getting stopped and unloaded by these lines of code. Hence why we still need to stop and unload the process but not necessarily run osascript since both systemwide & single user are accessing inside this condition. But I might be wrong I'll check to be sure